core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -887,10 +887,15 @@ class ReplicaManager(val config: KafkaConfig,
       val partitionFetchSize = fetchInfo.maxBytes
       val followerLogStartOffset = fetchInfo.logStartOffset
 
-      brokerTopicStats.topicStats(tp.topic).totalFetchRequestRate.mark()
-      brokerTopicStats.allTopicsStats.totalFetchRequestRate.mark()
-
       try {
+        if (allPartitions.contains(tp)) {
+           brokerTopicStats.topicStats(tp.topic).totalFetchRequestRate.mark()
+           brokerTopicStats.allTopicsStats.totalFetchRequestRate.mark()
+        } else { 
+          info(s"Delayed fetch for deleted partition $tp")
```

Comment:
We should not do this. `readFromLocalLog` should not encode information about its callers, it should be the other way around.

Reply:
Thanks, the log message is bad !